Fix SimpleSmartAnswer UI issues in Publisher

gclssvglx opened the pull request:
## What

Model validation error messages in Publisher are currently not being displayed to users.

We should make sure that - whether the user has JavaScript enabled or not - all model validation error messages are shown to the user and the appropriate field and field label are highlighted. 

This PR improves the `SimpleSmartAnswer` format's `Node` UI by showing model validation error messaging, and fixing style/formatting issues with `Options`. 

## Why

The user should be able to diagnose and correct the issue quickly and easily.

[Trello](https://trello.com/c/HTrLSx0G/549-fix-simplesmartanswer-ui-issues-in-publisher)

## Screenshots

### Before
![before](https://user-images.githubusercontent.com/44037625/128729377-3cb80a68-5583-419a-b16f-6d2851201be2.png)

### After

![Screenshot 2021-08-11 at 12 07 46](https://user-images.githubusercontent.com/44037625/129018953-49350fc6-20a4-4d8d-a842-aecc5cc9cd7b.png)

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

kevindew reviewed (commented):
Looks great, thanks for this.

My only comment is that we should replace the 3 chunks of error lists with the `form_errors` helper once we've merged https://github.com/alphagov/publisher/pull/1478 and rebased this on it.

gclssvglx commented:
> Looks great, thanks for this.
> 
> My only comment is that we should replace the 3 chunks of error lists with the `form_errors` helper once we've merged #1478 and rebased this on it.

Rebased and updated to use the `FormHelper` - thanks

kevindew commented on app/views/simple_smart_answers/_node.html.erb:
```diff
@@ -4,10 +4,14 @@
       <%= f.text_field :title, placeholder: "The title of the outcome", class: "node-title form-control" %>
-    </span>
+    </div>
+    <% if f.object.errors.has_key?(:title) %>
+      <%= form_errors(f.object.errors[:title]) %>
+    <% end %>
```
```suggestion
    <%= form_errors(f.object.errors[:title]) %>
```

It looks like [we have a convention already](https://github.com/alphagov/publisher/blob/a6438219a710f8dc6ddb23fd59eb9db0f7c71000/app/helpers/form_helper.rb#L18) to call this helper whether there are errors or not.

kevindew replied:
Same thing on the other two instances.

gclssvglx replied:
Yes, sorry.  Fixed

kevindew reviewed (commented):
Thanks
